v2/comments.go

```diff
@@ -249,12 +246,17 @@ type TypedTag struct {
 	Value string
 }
 
+// Arg represents a typed argument
 type Arg struct {
-	Name  string
+	// Name is the name of a named argument. This is zero-valued for positional arguments.
+	Name string
+	// Value is the string value of an argument. It has been validated to match the Type.
 	Value string
-	Typ   ArgType
+	// Type is the type of the Value.
+	Type ArgType
 }
```

Comment:
Is this string + type representation sufficient?  We could take this further and have types like:

```go
type String string

type (String) Type() ArgType {
  return TypeString
}
```

@aaron-prindle for thoughts on processing this downstream.